buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts

```diff
@@ -13,6 +13,15 @@ testlogger {
 
 tasks.withType<Test>().configureEach {
    useJUnitPlatform()
+   // Pass "kotest.*" system properties from the Gradle invocation to the test launcher.
+   // https://junit.org/junit5/docs/current/user-guide/#running-tests-config-params
+   for ((name, value) in System.getProperties()) {
+      if (name is String && name.startsWith("kotest.")) {
+         value as String
+         inputs.property(name, value)
+         systemProperty(name, value)
+      }
+   }
```

Comment:
I think that, unfortunately, iterating over all system properties will make configuration cache, and possibly build cache, be dependent on unrelated system properties. That means the caches can't be reused whenever an unrelated system property changes.

Fortunately Gradle has `providers.systemPropertiesPrefixedBy("kotest")`.

There's also a special way of passing in system properties to workaround [Gradle not fully supporting the Provider API](https://github.com/gradle/gradle/issues/12247).

I assume that you'll be happy with me fixing and pushing these changes directly?

Reply:
I've actually thought about that, but was not aware of the above solution. So yes, go ahead by any means! Thank you!

Reply:
Thanks!

Is there a list of valid Kotest system properties?

Reply:
Not that I'm aware of.

Reply:
This one: https://github.com/kotest/kotest/blob/0df229949563cca89f97770eb0b6e554fe997b8b/kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineProperties.kt

Reply:
I've searched for call sites of `syspropOrEnv()`. All system properties are listed in `KotestEngineProperties.kt`, plus two isolated ones: `KOTEST_DEBUG`, `KOTEST_DEBUG_PATH`. That should be complete.

Reply:
There are these extra places with system properties without an environment counterpart:
* https://github.com/kotest/kotest/blob/397466e17774dc98e2f479965aad91b0670b92b6/kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/AssertionsConfig.kt
* https://github.com/kotest/kotest/blob/b72c3729e6a4142d448bfe016e4cb582ea44a7b9/kotest-common/src/commonMain/kotlin/io/kotest/common/intellij.kt

Reply:
Thanks!

I wanted to have a look because if any of the properties were constantly changing (e.g. a timestamp), or some other system-dependent property (like a file path that changes per machine), then we'd have to take some extra care to avoid registering them as task inputs and breaking Gradle up-to-date checks.

For now I'll not register the values as task inputs. That means that even if they change then Gradle might not re-run the test tasks, but that's the case already.

It would be nice to have these defined in a single place, and in such a way that the Kotest Gradle Plugin could provide a nice DSL to set them in the correct way.

Reply:
When we switch to tag-based testing, registering values as input might become essential: We'll oscillate between
`-Dkotest.tags=!Deprecated` and `-Dkotest.tags=Deprecated` in consecutive test runs. But we can postpone covering that until we really need it.